pybamm/parameters/parameter_values.py

```diff
@@ -338,9 +338,28 @@ def check_parameter_values(self, values):
         if "C-rate" in values:
             raise ValueError(
                 "The 'C-rate' parameter has been deprecated, "
-                "use 'Current function [A]' instead. The cell capacity can be accessed "
-                "as 'Cell capacity [A.h]', and used to calculate current from C-rate."
+                "use 'Current function [A]' instead. The Nominal "
+                "cell capacity can be accessed as 'Nominal cell "
+                "capacity [A.h]', and used to calculate current from C-rate."
             )
+        if "Cell capacity [A.h]" in values:
+            if "Nominal cell capacity [A.h]" in values:
+                raise ValueError(
+                    "both 'Cell capacity [A.h]' and 'Nominal cell capacity [A.h]' "
+                    "provided in values. The 'Cell capacity [A.h]' notation will be "
+                    "deprecated in the next release so 'Nominal cell capacity [A.h]' "
+                    "should be used instead."
+                )
+            else:
+                values["Nominal cell capacity [A.h]"] = values["Cell capacity [A.h]"]
+                warnings.warn(
```

Comment:
You need to pass `DeprecationWarning` as an argument (see below) otherwise it raises a generic warning and that's why the test fails.

```python3
                warnings.warn(
                    "the 'Cell capacity [A.h]' notation will be "
                    "deprecated in the next release, as it has now been renamed "
                    "to 'Nominal cell capacity [A.h]'. Simulation will continue "
                    "passing the 'Cell capacity [A.h]' as 'Nominal cell "
                    "capacity [A.h]' (it might overwrite any existing definition "
                    "of the component)",
                    DeprecationWarning,
                )
```

Reply:
Yes, my bad, I didn't run the tests on my side. Running them now for confirmation.

Reply:
After adding this and running the tests, it gives me an error
```
pybamm.ParameterValues({"Cell capacity [A.h]": 0})
AssertionError: "Cell capacity [A.h]" does not match "the 'Cell capacity [A.h]' notation will be deprecated in the next release, as it has now been renamed to 'Nominal cell capacity [A.h]'. Simulation 
will continue passing the 'Cell capacity [A.h]' as 'Nominal cell capacity [A.h]' (it might overwrite any existing definition of the component)"
```
How can I resolve this?

Reply:
I don't know... Maybe try passing only "Cell capacity". Also, I don't think it is the source of any issue, but might be good to define the capacities in the test to be 1 rather than 0, as a 0 capacity could potentially lead to further issues.

Reply:
I tried that but then it shows the "Deprecation warning not triggered" error. I am looking into it, if you find the issue please let me know. Thanks

Reply:
Have you implemented my comment above about adding `DeprecationWarning` when you raise the warning? That should fix the issue about "Deprecation warning not triggered"

Reply:
Yes I did that, it does not give me the "Deprecation warning not triggered" with "Cell capacity [A.h]" but it does give me the same error with "Cell capacity".

Reply:
Can you push your recent changes so I can see the full error messages from the automatic tests?

Reply:
Yes sure, I was trying to use ```values=```, but that didn't work out as well.

Reply:
I tried it locally and it seems the issue is with `[A.h]`. It seems that this runs fine:
```python3
        # Cell capacity [A.h] deprecated
        with self.assertRaisesRegex(ValueError, "Cell capacity"):
            pybamm.ParameterValues({"Cell capacity [A.h]": 1,
                                    "Nominal cell capacity [A.h]": 1})
        with self.assertWarnsRegex(DeprecationWarning, "Cell capacity"):
            pybamm.ParameterValues({"Cell capacity [A.h]": 1})
```

Reply:
Thank you so much!